vnext/JSI/Shared/ChakraJsiRuntime_edgemode.cpp

```diff
@@ -1,40 +1,50 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-#include "ChakraRuntime.h"
+#include <ChakraRuntimeFactory.h>
+#include <SystemChakraRuntime.h>
 
 #include "ByteArrayBuffer.h"
 #include "Unicode.h"
 
-#if !defined(CHAKRACORE)
-#include <jsrt.h>
+// From <chakrart.h>
+STDAPI_(JsErrorCode)
+JsStartDebugging();
 
 namespace Microsoft::JSI {
 
-void ChakraRuntime::setupNativePromiseContinuation() noexcept {
+#if defined(USE_EDGEMODE_JSRT) && !defined(CHAKRACORE)
+/*static*/ void ChakraRuntime::initRuntimeVersion() noexcept {}
+#endif
+
+SystemChakraRuntime::SystemChakraRuntime(ChakraRuntimeArgs &&args) noexcept : ChakraRuntime(std::move(args)) {
+  this->Init();
+}
+
+void SystemChakraRuntime::setupNativePromiseContinuation() noexcept {
   // NOP
 }
 
-void ChakraRuntime::startDebuggingIfNeeded() {
+void SystemChakraRuntime::startDebuggingIfNeeded() {
   if (runtimeArgs().enableDebugging)
     JsStartDebugging();
 }
 
-void ChakraRuntime::stopDebuggingIfNeeded() {
+void SystemChakraRuntime::stopDebuggingIfNeeded() {
   // NOP AFAIK
 }
 
-void ChakraRuntime::initRuntimeVersion() noexcept {
-  // NOP
-}
-
-std::unique_ptr<const facebook::jsi::Buffer> ChakraRuntime::generatePreparedScript(
+std::unique_ptr<const facebook::jsi::Buffer> SystemChakraRuntime::generatePreparedScript(
     const std::string &sourceURL,
     const facebook::jsi::Buffer &sourceBuffer) noexcept {
   const std::wstring scriptUTF16 =
       Microsoft::Common::Unicode::Utf8ToUtf16(reinterpret_cast<const char *>(sourceBuffer.data()), sourceBuffer.size());
 
+#ifdef CHAKRACORE
```

Comment:
It looks strange here - the file supposed to be about the System Chakra.

Reply:
RNW Desktop always compiles with `CHAKRACORE` defined, yet, it needs to be able to instantiate this class if `ForceSystemChakra` is defined.

Hence, this code needs to be compiled with `CHAKRACORE` on.